libcontainer/rootfs_linux.go

```diff
@@ -572,6 +572,11 @@ func pivotRoot(rootfs, pivotBaseDir string) error {
 	if err != nil {
 		return fmt.Errorf("can't create pivot_root dir %s, error %v", pivotDir, err)
 	}
+	defer func() {
+		if errVal := os.Remove(pivotDir); errVal != nil && err == nil {
+			err = errVal
```

Comment:
@crosbymichael kk I was thinking just add the text in the case where err!=nil but I see this could be a bad pattern over the long haul. Do you ever do something like add a warning message when there is both an error and some possible cleanup to do?